### DISCUSSION TITLE
Nillion integration into Blockchain

### DISCUSSION THREAD

**Discussion by muhammadaus** [NillionNetwork/discussions#102] [Q&A]
Hi, I want to know how the workflow in Nillion looks like so the output can be passed to L1 Blockchains such as Ethereum Mainnet.
Is there any chart that I can refer to or at least explanation on how outputs can be integrated into an Ethereum smart contract?

I came from the docs by the way.

**Comment by lapets** [comment]
Thanks for the great question! It is in our roadmap to add some examples, templates, and possibly libraries to support this flow. But we'd love to understand what you have in mind and help. There are two situations/flows that we can imagine:
* **Public Output**: If you are okay reconstructing a secret output in one location (or if the output is simply public), you can run one of the clients in a cloud-based VM, locally hosted environment, or in your user's browser to obtain the output shares, reconstruct them, and pass them along to the L1.
* **Secret Output**: If you would like the raw shares of the secret output to be reconstructed on the L1, we can provide some guidance on the right way to structure your Nada program and how a smart contract can reconstruct the shares. However, at this time, you will be responsible for providing some additional decentralized infrastructure that sits between the Nillion Network and the L1.
